pkg/kscmd/debug.go

```diff
@@ -80,25 +68,24 @@ func StartDebugContainer(config SquashConfig) (*v1.Pod, error) {
 		}
 		survey.AskOne(prompt, &confirmed, nil)
 		if !confirmed {
-			return &v1.Pod{}, errors.New("user aborted")
+			return nil, errors.New("user aborted")
 		}
 	}
 
 	dbgpod, err := dp.debugPodFor(debugger, dbg.Pod, dbg.Container.Name)
 	if err != nil {
-		return &v1.Pod{}, err
+		return nil, err
 	}
 	debuggerPodNamespace := dp.config.Namespace
 	// create namespace. ignore errors as it most likely exists and will error
 	dp.getClientSet().CoreV1().Namespaces().Create(&v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: debuggerPodNamespace}})
 
 	createdPod, err := dp.getClientSet().CoreV1().Pods(debuggerPodNamespace).Create(dbgpod)
 	if err != nil {
-		return &v1.Pod{}, err
+		return nil, err
 	}
 
-	// TODO: we may be able to delete with DebugServer. see TODO below
-	if (!dp.config.DebugServer) && (!config.NoClean) {
```

Comment:
why was this condition changed?

Reply:
good catch, trying to trim down on the redundant flags, but I forgot to apply the NoClean flag to the in-cluster config, fixed with https://github.com/solo-io/squash/pull/73/commits/37f3e2e11569103e081440903c20b4ab88deef31